code-challenges.test.js

```diff
@@ -29,23 +59,129 @@ const colors2 = [
 
+// GREEN 
+// PSEUDOCODE
+// input: const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+// const colors2 = [
+//   "chartreuse",
+//   "indigo",
+//   "periwinkle",
+//   "ochre",
+//   "aquamarine",
+//   "saffron"
+// ]
+// output: an array that removes the first item from the array and shuffles the remaining content.
+
+// THE PROCESS:
+// create a function callre removeAndShuffle that takes an array as an argument 
+// create a new variable to remove the first item from the array(.splice)
+// make your .splice arguments starting at the zeroth index position and to remove the first element after the index start 
+// sort the remaining elements(.sort) and randomize the arrangement(Math.random)
+// return a shuffled array 
+
+
+const removeAndShuffle = (arr) => {
+  const removedItem = arr.splice(0, 1)
+  arr.sort(() => Math.random() - 0.5)
+  return arr
+}
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+
+// RED 
+describe("tallyVotes", () => {
+  const votes1 = { upVotes: 13, downVotes: 2 }
+  const votes2 = { upVotes: 2, downVotes: 33 }
+  const votes3 = { upVotes: 8, downVotes: 8 }
+
+  it("returns the correct tally for all votes", () => {
+    expect(tallyVotes(votes1)).toBe(11)
+    expect(tallyVotes(votes2)).toBe(-31)
+    expect(tallyVotes(votes3)).toBe(0)
+  })
+})
+
+// ReferenceError: tallyVotes is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.185 s, estimated 1 s
+
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
 
+// GREEN 
+// PSEUDOCODE 
+// input: an object that contains up votes and down votes
+// output: the end tally.
+
+// THE PROCESS:
+// create a function called tallyVotes that will take an object as an argument.
+// subtract the value of the downVotes property from the value of the upVotes property using math operators.
+// return the difference as the output of the function.
+
+const tallyVotes = (votes) => {
+  return votes.upVotes - votes.downVotes;
+}
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
+// RED 
+describe("mergeArrays", () => {
+  it("merges two arrays and removes duplicates", () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    const mergedArray = mergeArrays(dataArray1, dataArray2)
+    expect(mergedArray).toHaveLength(7)
```

Comment:
Nice added assertion for the test! Quick note: storing the function call within another variable is an unnecessary step!